urlb879d9e865_ogotalski_elevator_tgz-pJ8-ElevatorTaskJ8/src/by/epam/lab/view/LogViewFrame.java

```diff
@@ -0,0 +1,43 @@
+package by.epam.lab.view;
+
+import java.awt.Dimension;
+import java.awt.HeadlessException;
+import java.io.FileReader;
+import java.io.IOException;
+
+import javax.swing.BoxLayout;
+import javax.swing.JFrame;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+
+public class LogViewFrame extends JFrame {
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 1L;
+	private static final String LOGFILE_NAME = "logfile.log";
+	private static final Dimension DEFAULT_SIZE = new Dimension(1000, 800);
+
+	public LogViewFrame() throws HeadlessException {
+		super();
+		this.setTitle("Elevator Task");
+		this.setSize(DEFAULT_SIZE);
+		setDefaultCloseOperation(DISPOSE_ON_CLOSE);
+		setLayout(new BoxLayout(getContentPane(), BoxLayout.Y_AXIS));
+		JTextArea logArea = new JTextArea();
+		FileReader reader;
+		try {
+			reader = new FileReader(LOGFILE_NAME);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *reader*. Currently, there are execution paths that do not contain closure statements, for example, when JTextComponent.read throws an exception. Ensure *reader* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).